packages/dashboard-backend/src/devworkspaceClient/services/gitConfigApi/index.ts

```diff
@@ -37,13 +37,13 @@ export class GitConfigApiService implements IGitConfigApi {
    */
   public async read(namespace: string): Promise<api.IGitConfig> {
     try {
-      const response = await this.coreV1API.readNamespacedConfigMap(GITCONFIG_CONFIGMAP, namespace);
+      const response = await this.coreV1API.readNamespacedSecret(GITCONFIG_SECRET, namespace);
 
       return this.toGitConfig(response.body);
     } catch (error) {
       if (helpers.errors.isKubeClientError(error) && error.statusCode === 404) {
-        // Create gitconfig configmap if it does not exist
-        return this.createGitConfigMap(namespace);
+        // Create gitconfig secret if it does not exist
+        return this.createGitConfigSecret(namespace);
       }
```

Comment:
I'm wondering how this change would be working from the backward compatibility perspective?
e.g. if there is a cm with gitconfig once the new version of the project is rolled out would the gitconfig be blank?